MapboxCoreNavigation/RouteController.swift

```diff
@@ -852,8 +857,11 @@ extension RouteController: CLLocationManagerDelegate {
         guard let userSnapToStepDistanceFromManeuver = userSnapToStepDistanceFromManeuver else { return }
         guard let spokenInstructions = routeProgress.currentLegProgress.currentStepProgress.remainingSpokenInstructions else { return }
         
+        // Always give the first voice announcement when beginning a leg.
+        let firstInstructionOnFirstStep = routeProgress.currentLegProgress.stepIndex == 0 && routeProgress.currentLegProgress.currentStepProgress.spokenInstructionIndex == 0
```

Comment:
This worries me... 🤔 When we start hard-coding `index` in general, it means there's a bigger issue somewhere.

Reply:
Yep, https://github.com/mapbox/mapbox-navigation-ios/issues/901

Reply:
@bsudekum can we use the `distanceTraveled` on each step to determine when we are at the beginning of a step? For instance, if we are the beginning of a step, the distanceTraveled should be `zero` right?

Reply:
@bsudekum can we use the `distanceTraveled` on each step to determine when we are at the beginning of a step? For instance, if we are the beginning of a step, the `distanceTraveled` should be `zero` right?

Reply:
`distanceTraveled` is never guaranteed to be zero. In fact, it's only ever zero when you are behind the first coordinate.